ross/fluid_flow/fluid_flow_geometry.py

```diff
@@ -232,10 +233,40 @@ def calculate_eccentricity_ratio(modified_s):
         -(4 + (np.pi ** 2) * (modified_s ** 2)),
         1,
     ]
-    roots = np.roots(coefficients)
-    for i in range(0, len(roots)):
-        if 0 <= roots[i] <= 1:
-            return np.sqrt(roots[i].real)
+    roots = list(np.roots(coefficients))
+    roots2 = roots.copy()
+    for i in roots2:
+        if not np.isreal(i):
+            roots.remove(i)
+    del roots2
+
+    roots.sort()
+    for i in range(len(roots)):
+        roots[i] = roots[i].real
```

Comment:
`np.isreal`  checks for the imaginary part being exactly zero. Couldn't it be a problem if a root has a very small imaginary part, like `1e-16` due numerical truncation error?
```python
>>> np.isreal([1 + 1e-16j])
array([False])

>>> np.isreal([1 + 0j])
array([True])
```

Considering it's not a problem, you could rewrite this code like:
```python
    roots = np.roots(coefficients)
    roots = np.sort(roots[np.isreal(roots)].real)
```
Working with numpy array is more efficient than with pure lists.

Reply:
You're right. This way I can reach the same result.